pkg/utils/vdpa_provider.go

```diff
@@ -25,5 +28,18 @@ func GetVdpaProvider() VdpaProvider {
 }
 
 func (defaultVdpaProvider) GetVdpaDeviceByPci(pciAddr string) (vdpa.VdpaDevice, error) {
-	return vdpa.GetVdpaDeviceByPci(pciAddr)
+	// the govdpa library requires the pci address to include the "pci/" prefix
+	fullPciAddr := "pci/" + pciAddr
+	vdpaDevices, err := vdpa.GetVdpaDevicesByPciAddress(fullPciAddr)
+	if err != nil {
+		return nil, err
+	}
+	numVdpaDevices := len(vdpaDevices)
+	if numVdpaDevices == 0 {
+		return nil, fmt.Errorf("no vdpa device associated to pciAddress %v", pciAddr)
+	}
+	if numVdpaDevices > 1 {
```

Comment:
Is this a valid use-case?
should we always return the first one?

Reply:
Adrian asked the same question, please check comments above. Basically, there is only one vdpa device per pci-address in SRIOV. Anyway, the govdpa interface is more general and future proof for when sub-functions will be supported.

Reply:
Good that we ask the same questions :P